pkg/kubectl/cmd/annotation.go

```diff
@@ -124,7 +132,7 @@ func (o *AnnotateOptions) Complete(f *cmdutil.Factory, args []string, out io.Wri
 			return fmt.Errorf("all resources must be specified before annotation changes: %s", s)
 		}
 	}
-	if len(o.resources) < 1 {
+	if len(o.resources) < 1 && len(o.filenames) == 0 {
```

Comment:
@jlowdermilk Yes, if the user specifies both TYPE NAME and -f FILENAME,   the builder returns a user-friendly error message, that is "when paths, URLs, or stdin is provided as input, you may not specify a resource by arguments as well"